cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultDataTracerFactory.java

```diff
@@ -28,14 +29,16 @@
 public class DefaultDataTracerFactory implements DataTracerFactory {
 
   private final PreviewDataPublisher publisher;
+  private final CConfiguration cConf;
 
   @Inject
-  public DefaultDataTracerFactory(PreviewDataPublisher publisher) {
+  public DefaultDataTracerFactory(PreviewDataPublisher publisher, CConfiguration cConf) {
     this.publisher = publisher;
+    this.cConf = cConf;
   }
 
   @Override
   public DataTracer getDataTracer(ApplicationId applicationId, String tracerName) {
-    return new DefaultDataTracer(applicationId, tracerName, publisher);
+    return new DefaultDataTracer(applicationId, tracerName, publisher, this.cConf);
```

Comment:
minor nit: remove `this.` usually `this.` is only added when the member variable name conflicts with the method parameter name.